csm_web/frontend/src/components/course/ManageLabelsModal.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useState } from "react";
+import { fetchJSON, fetchWithMethod, HTTP_METHODS } from "../../utils/api";
+import LoadingSpinner from "../LoadingSpinner";
+import Modal from "../Modal";
+import { Label } from "../../utils/types";
+import Pencil from "../../../static/frontend/img/pencil.svg";
+import Trash from "../../../static/frontend/img/trash-alt.svg";
+import Exit from "../../../static/frontend/img/check_circle.svg";
+
+interface ManageLabelsModalProps {
+  closeModal: () => void;
+  courseId: number;
+  reloadSections: () => void;
+  title: string | boolean;
+}
+
+/**
+ * Modal that coords use to export a csv of student emails in selected courses.
+ */
+export const ManageLabelsModal = ({
+  closeModal,
+  courseId,
+  reloadSections,
+  title
+}: ManageLabelsModalProps): React.ReactElement => {
+  /**
+   * Map of course id to course name.
+   */
+  const [courseLabels, setCourseLabels] = useState<Label[]>([]);
+
+  const [currCourseLabels, setCurrCourseLabels] = useState<Label[]>([]);
+
+  const [newLabelKey, setNewLabelKey] = useState(-1);
+
+  // fetch all labels upon first mount
+  useEffect(() => {
+    fetchJSON(`/courses/${courseId}/labels`).then(data => {
+      setCourseLabels(data.sort((a: Label, b: Label) => a.id - b.id));
+      setCurrCourseLabels(data.sort((a: Label, b: Label) => a.id - b.id));
+    });
+  }, []);
+
+  const removeLabel = (id: number) => {
+    const tempLabels = courseLabels.filter((label: Label) => label.id !== id);
+    setCourseLabels(tempLabels);
+  };
+
+  const editRow = (id: number, name: string, description: string, showPopup: boolean) => {
+    const tempLabels = courseLabels.filter((label: Label) => label.id !== id);
+    const oldLabel = courseLabels.find((label: Label) => label.id === id);
+    const newLabel = { ...oldLabel, name: name, description: description, showPopup: showPopup } as Label;
+    let newLabels = [...tempLabels, newLabel];
+    const positiveLabelsSorted = newLabels.filter((label: Label) => label.id >= 0).sort((a, b) => a.id - b.id);
+    const negativeLabelsSorted = newLabels.filter((label: Label) => label.id < 0).sort((a, b) => b.id - a.id);
+    newLabels = [...positiveLabelsSorted, ...negativeLabelsSorted];
+    setCourseLabels(newLabels);
+  };
+
+  const handleNewRow = () => {
+    const course = courseLabels[0].course;
```

Comment:
This line assumes that courseLabels has elements, which is causing an access error when a course has no labels